README.md

```diff
@@ -63,35 +63,32 @@ C->S: Request {procedure_id, payload}
 S->C: Response {message_id, payload}
 ```
 
-#### Getting an async stream
+#### Getting an async stream (closed by client)
 
 ```sequence
 participant Scene (client) as C
 participant Kernel (server) as S
 C->S: Request {message_id}
-S->C: Response {message_id,streaming=true}
+S->C: Response {message_id,streaming=true,seqId=0}
 C->C: Generate async iterator for {message_id}
-S-->C: StreamMessage {message_id,payload}
-S-->C:
-S-->C:
+C->S: StreamMessage {ack=true,message_id,seqId=0}
+note over C: Ask for a new item to be generated using ack=true
+S-->C: StreamMessage {message_id,payload,seqId=1}
+C->S: StreamMessage {ack=true,message_id,seqId=1}
+note over C: Close the message by responding\nthe last ACK with ack=true,closed=true
+S-->C: StreamMessage {message_id,payload,seqId=2}
+C->S: StreamMessage {ack=true,message_id,seqId=2,closed=true}
+S->S: Close async Generator
+C->C: Close async Iterator
```

Comment:
should this `close async iterator` be here?

Reply:
I mean the `C->C: Close async Iterator` is duplicated and I think the last one is the right.

Reply:
the server has a generator, the client has an iterator. I think it is sound